### PR TITLE
extensible enum for blob servicve

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-blobs/CMakeLists.txt [hunk 1]
```diff
@@ -49,6 +49,7 @@ set(
     src/blob_batch_client.cpp
     src/blob_client.cpp
     src/blob_container_client.cpp
+	src/blob_rest_client.cpp
```
Comment (katmsft) [inline sdk/storage/azure-storage-blobs/CMakeLists.txt:52]: tab?

**Review by katmsft — CHANGES_REQUESTED** [review]
Please fix CI.

File: sdk/storage/azure-storage-blobs/CMakeLists.txt (hunk 1 above)
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/CMakeLists.txt:52]:
```suggestion
    src/blob_rest_client.cpp
```

**Review by katmsft — APPROVED** [review]

**Pull request merged**